src/query-builder/SelectQueryBuilder.ts

```diff
@@ -1962,4 +1973,20 @@ export class SelectQueryBuilder<Entity> extends QueryBuilder<Entity> implements
         return this.queryRunner || this.connection.createQueryRunner("slave");
     }
 
+    /**
+     * Gets the selected column from the selection. Is not always the same as the selection itself:
+     * 'DISTINCT ON (foo.bar) foo.bar' has the columnPath of 'foo.bar'
+     */
+    protected getSelectionWithColumnPath(selection: string) {
+        let columnPath: string = selection;
+
+        if (/(distinct)/i.test(selection) && this.connection.driver instanceof PostgresDriver) {
```

Comment:
```suggestion
        if (this.connection.driver instanceof PostgresDriver && /(distinct)/i.test(selection)) {
```
`this.connection.driver instanceof PostgresDriver` is easier to check(performance) then regexp, so it should be probably as first condition

Reply:
> ollija on Feb 28  Author
> I would've put this.connection.driver instanceof PostgresDriver-condition before the regexp, but that way some tests broke with this.connection being undefined

This was commented 2 months before your review. This is a very serious bug, makes MANY complex queries completely impossible to build via querybuilder.

Reply:
Looks like I've missed it. Thanks